activate

```diff
@@ -1,95 +1,218 @@
-# originally installed operating system version, the current version, the
-# machine vendor & product name. When the first activation message is sent, it
-# contains the machine serial number, but this is not sent again so cannot be
-# used for tracking purposes.
-# 
-# (C) 2014 Canonical Ltd.
-# (C) 2016 Endless Mobile Inc.
-# License: GPL v2 or later
-
-# https://home.endlessm.com/v1/activate
-# 
-# Arguments in the query string will be:
-# 
-#  image (the image name, from the xattr on /sysroot)
-#  release (from /etc/os-release)
-#  vendor (from DMI / device tree)
-#  product (from DMI / device tree)
-#  serial (from DMI / Endless mfgdata)
-#  live (true if this is a live USB, false if not)
-
-DIR="/var/lib/eos-phone-home"
-if [ ! -d "${DIR}" ] || [ ! -w "${DIR}" ]; then
-    echo "${DIR} missing or not writable"
-    exit 1
-fi
-
-# Give me a ping, Vasili. One ping only, please.
-ACTIVATEDFILE="${DIR}/activated"
-if [ -e "${ACTIVATEDFILE}" ]; then
-    echo "${ACTIVATEDFILE} exists, doing nothing"
-    exit 0
-fi
-
-SCRIPT="${EOS_PHONE_HOME_ACTIVATE_URL:-https://home.endlessm.com/v1/activate}"
-
-image=$(attr -q -g eos-image-version /sysroot 2>/dev/null ) || true
-if [ "${image}" != "" ]; then
-    image_root=$(attr -q -g eos-image-version / 2>/dev/null ) || true
-    if [ "${image_root}" != "" ]; then
-        image="${image_root}"
-    fi
-fi
-
-release=$(cat /etc/os-release | grep ^VERSION= | head -n1 | awk -F= '{ print $2 }') || true
-
-vendor=""
-product=""
-serial=""
-if [ -d "/sys/class/dmi/id" ]; then
-    vendor=$(< /sys/class/dmi/id/sys_vendor) || true
-    product=$(< /sys/class/dmi/id/product_name ) || true
-    serial=$(< /sys/class/dmi/id/product_serial) || true
-else
-    compat=$(cut -d '' -f1 < /proc/device-tree/compatible) || true
-    vendor=$(echo ${compat} | cut -d, -f1)
-    product=$(echo ${compat} | cut -d, -f2)
-    serial=$(< /sys/class/endless_mfgdata/entries/SSN) || serial=''
-fi
-
-# trim whitespace from strings
-vendor="${vendor%"${vendor##*[![:space:]]}"}"
-product="${product%"${product##*[![:space:]]}"}"
-serial="${serial%"${serial##*[![:space:]]}"}"
-
-# detect live system
-live="false"
-if grep -qw "endless.live" /proc/cmdline; then
-    image="true"
-fi
-
-# format as JSON
-vars="image release vendor product serial live"
-export ${vars}
-json="$(python -c 'import json, os, sys; print json.dumps(dict((v,os.environ[v]) for v in sys.argv[1:]))' ${vars})"
-
-# report in
-if ! out=$(wget -O - -q --method=PUT --body-data="${json}" "${SCRIPT}") ; then
-    echo "failed PUT to $SCRIPT with data: ${json}"
-    exit 1
-fi
-if ! ( echo "${out}" | python -c "import json, sys; sys.exit(0 if json.load(sys.stdin)['success'] == 'true' else 1)" 2>/dev/null ); then
-    echo "didn't get success from PUT to ${SCRIPT} with data: ${json}"
-    echo "got response: ${out}"
-    exit 1
-fi
-
-# touch file to record activation
-echo > "${ACTIVATEDFILE}"
+#!/usr/bin/env python3
 
```

Comment:
Can we reinstate the comments with the explanation of why the script exists and what it sends?